packages/quantic/force-app/main/default/lwc/quanticHeadlessLoader/quanticHeadlessLoader.js

```diff
@@ -158,20 +166,33 @@ function setComponentInitialized(element, engineId) {
     throw new Error('Fatal Error: Component was not registered before initialization');
   }
   component.initialized = true;
-  if (areAllComponentsInitialized(engineId)) {
-    debounceInitialSearch(engineId);
+  if (window.coveoHeadless[engineId].initializedCallback && areAllComponentsInitialized(engineId)) {
+    debounceInitializedCallback(engineId);
   }
 }
 
 /**
  * Returns headless engine promise.
  * @param {string} engineId The id of the engine.
  */
-async function getHeadlessEngine(engineId) {
-  if (!window.coveoHeadless[engineId].engine) {
-    window.coveoHeadless[engineId].engine = initEngine(engineId);
+function getHeadlessEnginePromise(engineId) {
+  if (!window.coveoHeadless[engineId].enginePromise) {
+    window.coveoHeadless[engineId].enginePromise = initEngine(engineId).then((engine) => {
+      window.coveoHeadless[engineId].bindings.engine = engine
+      return engine;
+    });
+  }
+  return window.coveoHeadless[engineId].enginePromise;
+}
+
+/**
+ * Returns bindings object for specified engineId.
+ * @param {string} engineId The id of the engine.
+ */
+function getHeadlessBindings(engineId) {
```

Comment:
Is this function required? It does not seem to be used anywhere.

I did not check how the bindings are used in atomic, but it is kind of weird to have the engine hold a bindings object that itself holds a reference on the engine. I feel we might be asking for trouble here.

Also, why are you wrapping the bindings in another object? Why not return the engine bindings directly?

Reply:
This ill be used by etienne in his breadcrumb manager.
The engine is not holding an instance of the bindings object, it goes `window.coveoHeadless[engineId].bindings.engine` so there's a usable reference to the engine that isn't a promise. and it's wrapped because if you look at the new interface definition in coveo.d.ts, you'll see there are other attributes planned.

Reply:
Yeah right, `window.coveoHeadless[engineId]` is not the engine itself, but more a collection of things required to build and initialize the engine.